src/deprecated/support.js

```diff
@@ -0,0 +1,11 @@
+define( [
+	"../var/support"
+], function( support ) {
+
+"use strict";
+
+support.trim = "" === "\uFEFF\xA0".trim();
```

Comment:
To be clear, I am arguing against the introduction of `support.trim` at +29 bytes (and really, even at +10).

Reply:
To me personally the biggest issue with size increases is they creep in, adding up to a meaningful increase over time. In this case, this danger does not exist as this is deprecated code that's already removed in `4.0.0-pre`. That makes +29 bytes more acceptable to me.

BTW, we could still make it smaller by dropping `\xA0` - this is actually trimmed properly by Android 4.0, `\uFEFF` is not, I just checked. That lowers the size increase to +24 bytes.

Reply:
Are you claiming that a real-life website, let alone a significant number of them, are spending so much time in `jQuery.trim` that an improvement from O(n²) to O(n) (with real effect of 240x for input with hundreds of medial whitespace characters and 23000x for input with tens of thousands, taking the OP 20 seconds down to about 1 ms) is not sufficient to address the overall experience? I'm not denying that native `String.prototype.trim` is astronomically faster still, but how would such sites even _get_ the input data fast enough for that further improvement to make a material difference?

That said, I'm not going to panic over 18 bytes in a legacy branch. If @mgol is in favor, that's fine with me.

Reply:
A real site, can and should, of course, call the native `trim()` method directly if its product requirements allows that, and could do so, at least for the handful of rare cases where this is known to make a significant difference.

While sites may be embedding a particular jQuery version of a variety of different reasons (e.g. blocked on migration of code, or offer jQuery API to other plugins, or browser support for _one_ of the legacy browsers, such as the case for Wikipedia where the JS payload supports IE11 and Android 5 but not IE9 or Android 4). It's quite likely a consumer could even adopt String-trim unconditionally in its code, if all its clients support ES5, even if jQuery itself maintains wider support.